File: ios/Runner/ExerciseCoaching/PoseDetectorToExerciseAdapter.swift
import Foundation
import MLKitPoseDetection

/// PoseDetectorProcessor ile ExerciseCoachProcessor arasındaki adapter.
/// Android'deki PoseDetectorToExerciseAdapter.java karşılığı (singleton pattern).
class PoseDetectorToExerciseAdapter {
    
    static let shared = PoseDetectorToExerciseAdapter()
    
    private var exerciseCoachProcessor: ExerciseCoachProcessor?
    private var enabled = false
    
    private init() {}
    
    /// ExerciseCoachProcessor'ı ayarla
    func setExerciseCoachProcessor(_ processor: ExerciseCoachProcessor?) {
        exerciseCoachProcessor = processor
        print("[PoseAdapter] ExerciseCoachProcessor set")
    }
    
    /// Adapter'ın etkin olup olmadığını ayarla
    func setEnabled(_ isEnabled: Bool) {
        enabled = isEnabled
        print("[PoseAdapter] Adapter enabled: \(enabled)")
    }
    
    /// Poz verisini exercise coach'a ilet
    func processPose(_ pose: Pose) {
        guard enabled, let processor = exerciseCoachProcessor else { return }
        
        do {
            processor.processPose(pose)
        } catch {
            print("[PoseAdapter] Error processing pose: \(error)")
        }
    }
}

File: mlkit_reference/android/app/src/main/java/com/google/mlkit/vision/demo/java/exercise_coaching/PoseDetectorToExerciseAdapter.java
package com.google.mlkit.vision.demo.java.exercise_coaching;

import android.util.Log;
import com.google.mlkit.vision.pose.Pose;

/**
 * PoseDetectorProcessor ile ExerciseCoachProcessor arasındaki adapter sınıfı.
 * Bu sınıf, algılanan pozları egzersiz koçluk sisteminize aktarmak için kullanılır.
 */
public class PoseDetectorToExerciseAdapter {
    private static final String TAG = "PoseAdapter";
    
    private static ExerciseCoachProcessor exerciseCoachProcessor;
    private static boolean enabled = false;
    
    /**
     * ExerciseCoachProcessor'ı ayarlar
     * 
     * @param processor ExerciseCoachProcessor
     */
    public static void setExerciseCoachProcessor(ExerciseCoachProcessor processor) {
        exerciseCoachProcessor = processor;
        Log.d(TAG, "ExerciseCoachProcessor set");
    }
    
    /**
     * Adapter'ın etkin olup olmadığını ayarlar
     * 
     * @param isEnabled Etkin durumu
     */
    public static void setEnabled(boolean isEnabled) {
        enabled = isEnabled;
        Log.d(TAG, "Adapter enabled: " + enabled);
    }
    
    /**
     * Poses are sent from the PoseDetectorProcessor to the ExerciseCoachProcessor
     * 
     * @param pose MLKit Pose nesnesi
     */
    public static void processPose(Pose pose) {
        if (!enabled || exerciseCoachProcessor == null || pose == null) {
            return;
        }
        
        try {
            exerciseCoachProcessor.processPose(pose);
        } catch (Exception e) {
            Log.e(TAG, "Error processing pose", e);
        }
    }
}
